src/java/org/apache/cassandra/service/accord/txn/TxnRangeRead.java

```diff
@@ -0,0 +1,351 @@
+                                                                                       receiver.accept(null, t);
+                                                                                   }
+                                                                               }
+
+                                                                               @Override
+                                                                               public long creationTimeNanos()
+                                                                               {
+                                                                                   return approxCreationTimeNanos;
+                                                                               }
+
+                                                                               @Override
+                                                                               public long startTimeNanos()
+                                                                               {
+                                                                                   return approxStartTimeNanos;
+                                                                               }
+
+                                                                               @Override
+                                                                               public String description()
+                                                                               {
+                                                                                   return command.toCQLString();
+                                                                               }
+                                                                           }
+        );
+    }
+
+    @Override
+    public Read slice(Ranges ranges)
+    {
+        return new TxnRangeRead(bytes(), cassandraConsistencyLevel, covering.slice(ranges, Slice.Minimal));
+    }
+
+    @Override
+    public Read intersecting(Participants<?> participants)
+    {
+        checkArgument(participants.domain() == Domain.Range, "Can only intersect with ranges");
+        return new TxnRangeRead(bytes(), cassandraConsistencyLevel, covering.intersecting(participants, Slice.Minimal));
+    }
+
+    @Override
+    public Read merge(Read other)
+    {
+        Ranges newCovering = ((TxnRangeRead)other).covering.union(UnionMode.MERGE_OVERLAPPING, covering);
+        return new TxnRangeRead(bytes(), cassandraConsistencyLevel, newCovering);
+    }
+
+    @Override
+    public Kind kind()
+    {
+        return Kind.range;
+    }
+
+    @Override
+    public long estimatedSizeOnHeap()
+    {
+        long size = EMPTY_SIZE;
+        size += ByteBufferUtil.estimatedSizeOnHeap(bytes());
+        size += AccordObjectSizes.ranges(covering);
+        return size;
+    }
+
+    @Simulate(with = MONITORS)
```

Comment:
I think this is unnecessary?

Reply:
It's necessary if you use synchronised and need the simulator to be aware of it, and to manage the locking (this is necessary if you want to simulate thread interleaving, or if other simulated actions might deadlock if the simulator doesn't know a monitor is being held). It shouldn't be necessary for any transitive monitor access (i.e. that within method calls). They need to be annotated themselves.

So I don't think it is necessary in this case.

Reply:
(It's also fairly harmless, but a bit confusing if you know what it means)